web/src/settings_ui.ts

```diff
@@ -34,7 +34,7 @@ export const strings = {
 export function do_settings_change(
     request_method: AjaxRequestHandler,
     url: string,
-    data: Parameters<AjaxRequestHandler>[0]["data"],
+    data: Omit<Parameters<AjaxRequestHandler>[0]["data"], "undefined"> | PatchRequestData,
```

Comment:
Reason:

1. `Omit`: the `data` parameter here can never be `undefined`. I have cross-checked !.
2.  `PatchRequestData` union : for when `request_method ` function is of type `channel.patch`